controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -160,6 +160,12 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(ctx context.Context
 		log.Error(err, "Error initializing enactment")
 	}
 
+	enactmentInstance, err := r.applyEnactmentDefaults(instance)
```

Comment:
Not related to this PR, but what is the reason for not returning an error when `initializeEnactment` fails.

Reply:
Actually, I think it is related now.
Previously, I suppose, we were okay with the enactment not created, as it didn't have any consequence for applying the policy desiredState.
But the current code passes the desiredState in the enactment, so if it happened to be nil, it would blow.

Reply:
[Updated](https://github.com/nmstate/kubernetes-nmstate/pull/793/commits/c1ec4a4fb04ba0f8c95825112b40c053a032b5c8)